bach/bach/series/utils/datetime_formats.py

```diff
@@ -0,0 +1,192 @@
+    '%V',  # ISO_8601_WEEK_NUMBER_OF_YEAR
+    '%u',  # ISO_8601_WEEKDAY_NUMBER
+
+    # time unit codes
+    '%H',  # HOUR24
+    '%I',  # HOUR12
+    '%k',  # HOUR24_PRECEDED_BY_A_SPACE
+    '%l',  # HOUR12_PRECEDED_BY_A_SPACE
+    '%M',  # MINUTE
+    '%S',  # SECOND
+    '%s',  # EPOCH
+    '%f',  # MICROSECOND
+
+    '%z',  # UTC_OFFSET
+    '%Z',  # TIME_ZONE_NAME
+
+    # format codes
+    '%R',  # HOUR_MINUTE
+    '%T',  # HOUR_MINUTE_SECOND
+
+    # special characters
+    '%n',  # NEW_LINE
+    '%t',  # TAB
+    '%%',  # PERCENT_CHAR
+}
+
+# https://www.postgresql.org/docs/current/functions-formatting.html#FUNCTIONS-FORMATTING-DATETIME-TABLE
+_C_STANDARD_CODES_X_POSTGRES_DATE_CODES = {
+    "%a": "Dy",
+    "%A": "FMDay",
+    "%w": "D",  # Sunday is 1 and Saturday is 7
+    "%d": "DD",
+    "%j": "DDD",
+    "%b": "Mon",
+    "%h": "Mon",
+    "%B": "FMMonth",
+    "%m": "MM",
+    "%y": "YY",
+    "%Y": "YYYY",
+    "%C": "CC",
+    "%Q": "Q",
+    "%D": "MM/DD/YY",
+    "%F": "YYYY-MM-DD",
+    "%G": "IYYY",
+    "%g": "IY",
+    "%V": "IW",
+    "%u": "ID",
+    "%H": "HH24",
+    "%I": "HH12",
+    "%M": "MI",
+    "%S": "SS",
+    "%f": "US",
+    "%z": "OF",
+    "%Z": "TZ",
+    "%R": "HH24:MI",
+    "%T": "HH24:MI:SS",
+}
+
+
+def parse_c_standard_code_to_postgres_code(date_format: str) -> str:
+    """
+    Parses a date format string from standard codes to Postgres date codes.
+    If c-standard code has no equivalent, it will remain in resultant string.
+
+    Steps to follow:
+        date_format = '%Y%m-%Y%m-%Y%m%d-%d'
+
+        Step 1: Get all unique groups of continuous c-codes,
+            groups are processed based on length in order to avoid
+            replacing occurrences of other groups (e.g %Y%m occurs in %Y%m%d, but both are different groups):
+                ['%Y%m%d', '%Y%m', '%d']
+
+        Step 2: Split initial dateformat by c-code groups
+
+        Step 3: For token from previous step:
+            If token is not a supported c-code, then treat the token as a literal and quote it.
+                (e.g Year: -> "Year:")
+            else:
+                1) Get individual codes from token
+                2) Replace each code with respective Postgres Code
+                3) Recreate group by joining all results from previous step with an empty literal.
+                    (e.g %Y%m%d -> YYYY""MM""DD
+
+    .. note:: We use double quotes to force TO_CHAR interpret an empty string as literals, this way
+            continuous date codes yield the correct value as intended
+            For example having '%y%Y':
+              TO_CHAR(cast('2022-01-01' as date), 'YY""YYYY') will generate '222022' (correct)
+              vs.
+              TO_CHAR(cast('2022-01-01' as date), 'YYYYYY') will generate '202222' (incorrect)
+
+    """
+
+    codes_base_pattern = '|'.join(_SUPPORTED_C_STANDARD_CODES)
+    grouped_codes_matches = re.findall(pattern=rf"(?P<codes>({codes_base_pattern})+)", string=date_format)
```

Comment:
```suggestion
    grouped_codes_matches = re.findall(pattern=rf"(?P<codes>(?:{codes_base_pattern})+)", string=date_format)
```
Then we only get a single group from the regex. and we don't need to discard the second one (line 137 and 140)

Reply:
hah worked! Thanks!